docs/en-us/cloneset.md

```diff
@@ -345,7 +345,114 @@ spec:
     paused: true
 ```
 
-### PreUpdate and PostUpdate
+### Lifecycle hook
 
-`PreUpdate` and `PostUpdate` can allow users to specify extra tasks before and after Pod update.
-This feature will be available in future release.
+Since v0.6.1, kruise has supported lifecycle hook for CloneSet.
+
+Each Pod managed by CloneSet has a clear state defined in `lifecycle.apps.kruise.io/state` label:
+
+- Normal
+- PreparingUpdate
+- Updating
+- Updated
+- PreparingDelete
+
+Lifecycle hook allows users to do something during Pod deleting and before/after in-place update.
+
+```golang
+type LifecycleStateType string
+
+// Lifecycle contains the hooks for Pod lifecycle.
+type Lifecycle struct {
+    // PreDelete is the hook before Pod to be deleted.
+    PreDelete *LifecycleHook `json:"preDelete,omitempty"`
+    // InPlaceUpdate is the hook before Pod to update and after Pod has been updated.
+    InPlaceUpdate *LifecycleHook `json:"inPlaceUpdate,omitempty"`
+}
+
+type LifecycleHook struct {
+    LabelsHandler     map[string]string `json:"labelsHandler,omitempty"`
+    FinalizersHandler []string          `json:"finalizersHandler,omitempty"`
+}
+```
+
+Examples:
+
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: CloneSet
+spec:
+
+  # define with finalizer
+  lifecycle:
+    preDelete:
+      finalizersHandler:
+      - example.io/unready-blocker
+    inPlaceUpdate:
+      finalizersHandler:
+      - example.io/unready-blocker
+
+  # or define with label
+  lifecycle:
+    inPlaceUpdate:
+      labelsHandler:
+        example.io/block-unready: "true"
+```
+
+#### State circulation
+
+![OpenKruise components](/img/docs/cloneset-lifecycle.png)
+
+- When CloneSet delete a Pod (including scale in and recreate update):
+  - Delete it directly if no lifecycle hook definition or Pod not matched preDelete hook
+  - Otherwise, CloneSet will firstly update Pod to `PreparingDelete` state and wait for user controller remove the label/finalizer and Pod not matched preDelete hook
```

Comment:
wait for user controller to remove the label/finalizer。 add "to"

> Pod not matched preDelete hook 
any missing sentence here ?